File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
from collections import deque

def solution(cards1, cards2, goal):
    answer = ''
    flag = True
    cards1 = deque(cards1)
    cards2 = deque(cards2)
    goal = deque(goal)
    
    while goal:
        word = goal.popleft()
        if cards1 and cards1[0] == word:
            cards1.popleft()
            continue
        
        if cards2 and cards2[0] == word:
            cards2.popleft()
            continue
        
        flag = False
        break

    return "Yes" if flag else "No"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    let answer = "Yes";
    for (let i = 0; i < goal.length; i++) {
        const target = goal[i]
        if (cards1[0] === target) {
            cards1.shift()
            continue
        }
        
        if (cards2[0] === target) {
            cards2.shift()
            continue
        }
        
        answer = "No"
        break
    }
    return answer;
}